src/test/java/com/example/javaOjt/services/AuthorServiceTest.java

```diff
@@ -77,4 +80,73 @@ void getAuthorById_withBooks_false() {
     Mockito.verify(authorRepository, Mockito.never()).getAuthorWithBookDTOsById(targetId);
     Mockito.verify(authorRepository, Mockito.times(1)).findById(Mockito.any());
   }
+
+  @Test
+  void getAuthorsResponse_noParams() {
+    // Mocking the Authors response
+    List<Author> mockAuthors = List.of(new Author());
+    mockAuthors.get(0).setId(1);
+    mockAuthors.get(0).setName("Test Author1");
+    Mockito.when(authorRepository.getAuthorsList(AuthorSortBy.ID, null, null))
+        .thenReturn(mockAuthors);
+    // Call the service method
+    GetAuthorsResponse response = authorService.getAuthorsList(null, null, null);
+    // Assertions to verify the response
+    Assertions.assertNotNull(response);
+    Assertions.assertEquals(mockAuthors.size(), response.getAuthorsSummaryList().size());
+    Assertions.assertEquals(mockAuthors.getFirst().getId(),
+        response.getAuthorsSummaryList().getFirst().id());
+    Assertions.assertEquals(mockAuthors.getFirst().getName(),
+        response.getAuthorsSummaryList().getFirst().name());
+    Mockito.verify(authorRepository, Mockito.times(1)).getAuthorsList(AuthorSortBy.ID, null, null);
+  }
+
+  @Test
+  void getAuthorsResponse_ids() {
+    int targetId1 = 1;
+    int targetId2 = 2;
+
+    // Mocking the Authors response
+    List<Author> mockAuthors = List.of(new Author(), new Author());
+    mockAuthors.get(0).setId(targetId1);
+    mockAuthors.get(1).setId(targetId2);
+    Mockito.when(
+            authorRepository.getAuthorsList(AuthorSortBy.ID, null, List.of(targetId1, targetId2)))
+        .thenReturn(mockAuthors);
+
+    // Call the service method
+    GetAuthorsResponse response = authorService.getAuthorsList(null, null,
+        List.of(targetId1, targetId2));
+
+    // Assertions to verify the response
+    Assertions.assertNotNull(response);
+    Assertions.assertEquals(mockAuthors.size(), response.getAuthorsSummaryList().size());
+    Assertions.assertEquals(mockAuthors.get(0).getId(),
+        response.getAuthorsSummaryList().get(0).id());
+    Assertions.assertEquals(mockAuthors.get(1).getId(),
+        response.getAuthorsSummaryList().get(1).id());
+    Mockito.verify(authorRepository, Mockito.times(1))
+        .getAuthorsList(AuthorSortBy.ID, null, List.of(targetId1, targetId2));
+  }
+
+  @Test
+  void getAuthorsResponse_invalidId() {
```

Comment:
```suggestion
  void getAuthorsResponse_notExistedId() {
```

ちなみに個人の意見ですが、このテストケースは不要だと考えています。

理由: service 層は単純に repository 層の結果を受け取って、ビジネスロジックで処理した結果を Controller 層に返す役割なので、id が存在するかどうかは service 層の関心事ではないと考えています

Reply:
確かに、おっしゃる通りです。